.openshift-ci/end.sh

```diff
@@ -8,63 +8,154 @@ source "$ROOT/scripts/ci/lib.sh"
+
+    post_process_test_results "${END_SLACK_FAILURE_ATTACHMENTS}"
+
+    if [[ "${OVERALL_JOB_OUTCOME}" == "${OUTCOME_FAILED}" ]]; then
+        (send_slack_failure_summary) || { echo "ERROR: Could not slack a test failure message"; }
+    fi
+}
+
+determine_an_overall_job_outcome() {
+    # Determin a useful overall job outcome based on state shared from prior steps.
+    # 'undefined' states mean the step did not run or openshift-ci canceled it.
+    # Note: in openshift-ci, if SHARED_DIR files are created or changed after
+    # cancelation that does not propagate.
+
+    combined="${CREATE_CLUSTER_OUTCOME:-undefined}-${JOB_DISPATCH_OUTCOME:-undefined}-${DESTROY_CLUSTER_OUTCOME:-undefined}"
+
+    info "Determining a job outcome from (cluster create-job-cluster destroy):"
+    info "${combined}"
+
+    case "${combined}" in
+        undefined-undefined-*)
+            # The job was interrupted before cluster create could complete. or
+            # openshift-ci had a meltdown. cluster destroy might still pass, fail or
+            # be canceled.
+            outcome="${OUTCOME_CANCELED}"
+            ;;
+        "${OUTCOME_FAILED}"-*-*)
+            # Track cluster create failures
+            outcome="${OUTCOME_FAILED}"
+            ;;
+        "${OUTCOME_PASSED}"-undefined-*)
+            # The job was interrupted before the test could complete. or somewhat
+            # less likely openshift-ci had a meltdown.
+            outcome="${OUTCOME_CANCELED}"
+            ;;
+        "${OUTCOME_PASSED}-${OUTCOME_FAILED}"-*)
+            outcome="${OUTCOME_FAILED}"
+            ;;
+        "${OUTCOME_PASSED}-${OUTCOME_PASSED}"-undefined)
+            # The job was interrupted before cluster destroy could complete, this is
+            # not ideal but we can rely on janitor to clean up, for actionableness
+            # track as a passing job.
+            outcome="${OUTCOME_PASSED}"
+            ;;
+        "${OUTCOME_PASSED}-${OUTCOME_PASSED}-${OUTCOME_FAILED}")
+            # Track cluster destroy failures as overall job failures.
+            outcome="${OUTCOME_FAILED}"
+            ;;
+        "${OUTCOME_PASSED}-${OUTCOME_PASSED}-${OUTCOME_PASSED}")
+            # Perfect score!
+            outcome="${OUTCOME_PASSED}"
+            ;;
+        *)
+            info "ERROR: unexpected state in end.sh: ${combined}"
+            outcome="${OUTCOME_FAILED}"
+            ;;
+    esac
+
+    export OVERALL_JOB_OUTCOME="${outcome}"
+    info "Overall job outcome: ${outcome}"
+}
+
+_JUNIT_CLUSTER_CLASS="Cluster"
+_JUNIT_CLUSTER_CREATE_DESCRIPTION="Create"
+_JUNIT_CLUSTER_DESTROY_DESCRIPTION="Destroy"
+
+# generate_cluster_junit() - generate junit records for cluster create & destroy
+# pass & fail.
+generate_cluster_junit() {
+    # Change the presentation for cluster types
+    local cluster_flavor_variant="${CLUSTER_FLAVOR_VARIANT:-unknown}"
+    if [[ ${#cluster_flavor_variant} -le 5 ]]; then
+        # uppercase GKE, ARO, ROSA, etc
```

Comment:
Let's avoid clever shortcuts like this and make this a case/esac that explicitly looks at the prefix.

Reply:
This is _just_ a slight presentation improvement for JIRA and Slack. I don't think it is worth the effort to enumerate and maintain all the flavors (which change separately from this repo) here.